### PR TITLE
Fixed cannnot call numpy on variable that requires grad error

### PR DESCRIPTION
**Pull request by Priyansi**
In lesson2-sgd, an error pops up, specifically, cannot call numpy on variable that requires grad. I have included a simple fix by changing x@a to x@a.detach().numpy(). This solves the error.

### CODE REVIEW

**Comment by Priyansi** [comment]
Can you please tell me why this failed?
